nexus/db-model/src/ipv4_nat_entry.rs

```diff
@@ -0,0 +1,93 @@
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use super::MacAddr;
+use crate::{
+    schema::{ipv4_nat_entry, ipv4_nat_version},
+    Ipv4Net, Ipv6Net, SqlU16, Vni,
+};
+use chrono::{DateTime, Utc};
+use omicron_common::api::external;
+use schemars::JsonSchema;
+use serde::Serialize;
+use uuid::Uuid;
+
+/// Values used to create an Ipv4NatEntry
+#[derive(Insertable, Debug, Clone)]
+#[diesel(table_name = ipv4_nat_entry)]
+pub struct Ipv4NatValues {
+    pub external_address: Ipv4Net,
+    pub first_port: SqlU16,
+    pub last_port: SqlU16,
+    pub sled_address: Ipv6Net,
+    pub vni: Vni,
+    pub mac: MacAddr,
+}
+
+/// Database representation of an Ipv4 NAT Entry.
+#[derive(Queryable, Debug, Clone, Selectable)]
+#[diesel(table_name = ipv4_nat_entry)]
+pub struct Ipv4NatEntry {
+    pub id: Uuid,
+    pub external_address: Ipv4Net,
+    pub first_port: SqlU16,
+    pub last_port: SqlU16,
+    pub sled_address: Ipv6Net,
+    pub vni: Vni,
+    pub mac: MacAddr,
+    pub version_added: i64,
+    pub version_removed: Option<i64>,
+    pub time_created: DateTime<Utc>,
+    pub time_deleted: Option<DateTime<Utc>>,
+}
+
+impl Ipv4NatEntry {
+    pub fn first_port(&self) -> u16 {
+        self.first_port.into()
+    }
+
+    pub fn last_port(&self) -> u16 {
+        self.last_port.into()
+    }
+}
+
+/// Database representation of an Ipv4 NAT Generation.
```

Comment:
Hey, it took me a while to figure this out -- this is a view of a sequence, not a table, and the `log_cnt` + `is_called` values are synthetically provided from CockroachDB.

That's worth documenting! I do not find that behavior obvious - I was trying to find out where these were used.

Related: Do we actually need to include `log_cnt` and `is_called` here? When we query for the Ipv4NatGen, we only read `last_value`, right?

Reply:
Yes, thus far we only care about `last_value`, I can delete the other two columns from the struct

Reply:
Actually, it seems that since we're not using this struct anymore thanks to some of the other changes that have been made :D